### PR TITLE
Ticket4913 muon begin end commands

### PR DESCRIPTION
**Pull request by JamesKingWork**
Add an upgrade step to check if an instrument uses the old style precmd and postcmd hooks.

### CODE REVIEW

File: src/upgrade_step_check_init_inst.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from src.file_access import FileAccess
+from src.local_logger import LocalLogger
+from src.upgrade_step import UpgradeStep
+from src.common_upgrades.utils.constants import CONFIG_ROOT
+import os
+# from io import open
```
Comment (DominicOram) [inline src/upgrade_step_check_init_inst.py]: Remove unused code

File: src/upgrade_step_check_init_inst.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        Recursively search folders for the search string.
+
+        Args:
+            folder (str): The folder to search through.
+            file_access (FileAccess): file access
+        
+        Returns: 0 if pre and post cmd methods in old style are not present; error message if they are.
+        """
+        root, dirs, files = os.walk(folder)
```
Comment (DominicOram) [inline src/upgrade_step_check_init_inst.py]: `walk` gives a generator so you can't unpack it like this

File: src/upgrade_step_check_init_inst.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        # Search files for pre and post cmd methods
+        file_search_return = self.search_files(files, root, file_access)
+        if file_search_return != 0:
+            return file_search_return
+        # Recurse to search in further directories
+        for directory in dirs:
+            directories_search_return = self.search_folder(os.path.join(root, directory), file_access)
+            if directories_search_return != 0:
+                return directories_search_return
```
Comment (DominicOram) [inline src/upgrade_step_check_init_inst.py]: `walk` is already recursive

File: src/upgrade_step_check_init_inst.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            file_access (FileAccess): file access
+        
+        Returns: 0 if pre and post cmd methods in old style are not present; error message if they are.
+        """
+        for file_name in files:
+            if file_name.startswith("init_"):
+                search_file = open(os.path.join(root, file_name))
+                search_file_contents = search_file.read()
+                if "precmd" in search_file_contents or "postcmd" in search_file_contents:
+                    return "Pre or post cmd methods found in {} convert to new style of inserting these methods".format(search_file.name)
```
Comment (DominicOram) [inline src/upgrade_step_check_init_inst.py]: Can you be a bit more explicit on how to do this or link to the wiki page that shows you how.

File: test/test_upgrade_step_check_init_inst.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+    def test_GIVEN_no_file_called_init_inst_WHEN_search_files_THEN_zero_returned(self):
+        # Arrange
+        file_names = ["init", "my_init", "another_file"]
+        root = "myfolder"
+        # Act and Assert
+        assert_that(self.upgrade_step.search_files(file_names, root, self.file_access),
+            equal_to(0), "File starting with init_inst is not in filenames, so no pre or post cmd would be found by genie")
+    
+    def test_GIVEN_no_file_called_init_inst_containing_precmd_WHEN_search_files_THEN_zero_returned(self):
```
Comment (DominicOram) [inline test/test_upgrade_step_check_init_inst.py]: This is exactly the same as the test above

File: src/upgrade_step_check_init_inst.py (hunk 2 above)
Comment (JamesKingWork) [reply src/upgrade_step_check_init_inst.py]: Ahh my bad, yes I should just use it the regular way, no idea why I decided to use it this way.

File: src/upgrade_step_check_init_inst.py (hunk 3 above)
Comment (JamesKingWork) [reply src/upgrade_step_check_init_inst.py]: Agreed, this was a bit of a dumb way to do this. Not sure my brain was working the day I wrote this.

File: src/upgrade_step_check_init_inst.py (hunk 4 above)
Comment (JamesKingWork) [reply src/upgrade_step_check_init_inst.py]: Link added which describes the pre and post command hook: https://github.com/ISISComputingGroup/ibex_user_manual/wiki/genie_python-Commands

File: test/test_upgrade_step_check_init_inst.py (hunk 5 above)
Comment (JamesKingWork) [reply test/test_upgrade_step_check_init_inst.py]: Yes, not sure what was meant by the test. Removed.

**Pull request merged**